### DISCUSSION TITLE
gjs-tools element not position right when adding transition for layout

### DISCUSSION THREAD

**Discussion by nguyenvoanhkhoa1** [GrapesJS/grapesjs#5404] [Q&A] [answered]
I am creating a custom new site layout, so I wrote the custom preview command. The behavior is ok, but when I add transition CSS for the layout like 2 sidebars and a topbar, the gjs-tools element position is not true when escape the preview mode.
The image below is when I entered preview mode and then escaped,
![image](https://github.com/GrapesJS/grapesjs/assets/55738921/7e0a7236-4280-4f46-a7ee-c83dd8161cd9)
Look like the gjs-tools element has an absolute position based on the screen but not the component that I hover or select. When I enter preview mode, the left sidebar has a negative margin-left, with the value as the width of the sidebar, then when escape preview mode, the margin will be 0, and the position left will be added for a value of the width of the sidebar
Is there any solution for it? If not, I will remove the transition behavior, then the gjs-tools will be normal.
Thank you.

**Comment by artf** [comment, marked as answer]
You have to run [`editor.refresh`](https://grapesjs.com/docs/api/editor.html#refresh) once the transition is completed.

**Reply by nguyenvoanhkhoa1** [reply]
Thank you. I've added editor.refresh() but it is not working, due to the animation taking time to complete, so I add setTimeout and it works now. Thank you for your help.
